consensus/README.md

```diff
@@ -0,0 +1,205 @@
+# Consensus Module <!-- omit in toc -->
+
+This README serves as a guide to the implementation of the [1.0 Pocket's Consensus Module Specification](https://github.com/pokt-network/pocket-network-protocol/tree/main/consensus). It is designed to provide insights into the structure and design of the codebase.
+
+## Table of Contents <!-- omit in toc -->
+
+- [Interface](#interface)
+- [Consensus Processes](#consensus-processes)
+  - [Leader Election](#leader-election)
+  - [Consensus Phases](#consensus-phases)
+  - [Block Generation](#block-generation)
+  - [Block Validation](#block-validation)
+  - [Consensus Lifecycle](#consensus-lifecycle)
+  - [State Sync](#state-sync)
+- [Implementation](#implementation)
+  - [Code Organization](#code-organization)
+- [Testing](#testing)
+  - [Running Unit Tests](#running-unit-tests)
+
+## Interface
+
+This module adheres to the interface outlined in `pocket/shared/modules/consensus_module.go`, which is derived from the above specification.
+
+## Consensus Processes
+
+This repository features an implementation of the HotStuff consensus algorithm. It facilitates the consensus process through a series of rounds. Staked validator nodes participate in the consensus process, with one node serving as the leader and the rest as replicas.
+
+### Leader Election
+
+A dedicated submodule handles the leader election process. The current configuration employs a deterministic round-robin leader election mechanism.
```

Comment:
Should we mention Algoran'd leader election (and potentially link to it too), explaining that it's the next step?